packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -281,3 +281,187 @@ def thread_request():
+        'delegate': {
+            'type': 'mesos-cni-port-mapper',
+            'excludeDevices': ['sprt-cni0'],
+            'chain': 'spartan-net',
+            'delegate': {
+                'type': 'bridge',
+                'bridge': 'sprt-cni0',
+                'ipMasq': True,
+                'isGateway': True,
+                'ipam': {
+                    'type': 'host-local',
+                    'subnet': '192.168.250.0/24',
+                    'routes': [
+                     # Reachability to the default `dcos` overlay subnet.
+                     {'dst': '9.0.0.0/8'},
+                     # Reachability to all private address subnet.
+                     {'dst': '10.0.0.0/8'},
+                     {'dst': '172.16.0.0/12'},
+                     {'dst': '192.168.0.0/16'}
+                    ]
+                }
+            }
+        }
+    }
+
+    log.info("spartan-net config:{}".format(json.dumps(spartan_net)))
+
+    # Application to deploy CNI configuration.
+    cni_config_app, config_uuid = test_helpers.marathon_test_app()
+
+    # Override the default test app command with a command to write the CNI
+    # configuration.
+    #
+    # NOTE: We add the sleep at the end of this command so that the task stays
+    # alive for the test harness to make sure that the task got deployed.
+    # Ideally we should be able to deploy one of tasks using the test harness
+    # but that doesn't seem to be the case here.
+    cni_config_app['cmd'] = 'echo \'{}\' > /opt/mesosphere/etc/dcos/network/cni/spartan.cni; sleep 10000'.format(
+        json.dumps(spartan_net))
+
+    log.info("App for setting CNI config: {}".format(json.dumps(cni_config_app)))
+
+    try:
+        dcos_api_session.marathon.deploy_app(cni_config_app, check_health=False)
+    except Exception as ex:
+        raise AssertionError("Couldn't install CNI config for `spartan-net`".format(json.dumps(cni_config_app))) from ex
+
+    # Get the host on which the `spartan-net` was installed.
+    cni_config_app_service = None
+    try:
+        cni_config_app_service = dcos_api_session.marathon.get_app_service_endpoints(cni_config_app['id'])
+    except Exception as ex:
+        raise AssertionError("Couldn't retrieve the host on which `spartan-net` was installed.") from ex
+
+    # We only have one instance of `cni_config_app_service`.
+    spartan_net_host = cni_config_app_service[0].host
+
+    # Launch the test-app on spartan-net, with a VIP.
+    server_vip_port = unused_port('spartanvip')
+    server_vip = '/spartanvip:{}'.format(server_vip_port)
+    server_vip_addr = 'spartanvip.marathon.l4lb.thisdcos.directory:{}'.format(server_vip_port)
+
+    # Launch the test_server in ip-per-container mode (user network)
+    server, test_uuid = test_helpers.marathon_test_app(
+        container_type=marathon.Container.MESOS,
+        healthcheck_protocol=marathon.Healthcheck.MESOS_HTTP,
+        network=marathon.Network.USER,
+        host_port=9080,
```

Comment:
Do you intend to access the service via hostIP:hostPort or directly via containerIP:containerPort? Because this will force you to hostIP:hostPort

Reply:
The `marathon_test_app` API set the `container_port` directly to the `host_port`. The API doesn't take a `container_port`. However, internally even though we specify a `host_port` it doesn't setup up `port-mapping` for USER networks.  So in affect the backend network-scope for user networks through this API will always be set to the overlay IP and not the host IP.